services/src/datasets/external/natur40.rs

```diff
@@ -0,0 +1,290 @@
+use std::path::Path;
+
+use crate::error::Error;
+use crate::util::parsing::string_or_string_array;
+use crate::{datasets::listing::DatasetListOptions, error::Result};
+use crate::{
+    datasets::{
+        listing::{DatasetListing, DatasetProvider},
+        storage::DatasetProviderDefinition,
+    },
+    error,
+    util::user_input::Validated,
+};
+use async_trait::async_trait;
+use futures::future::join_all;
+use gdal::DatasetOptions;
+use geoengine_datatypes::dataset::{DatasetId, DatasetProviderId, ExternalDatasetId};
+use geoengine_operators::engine::TypedResultDescriptor;
+use geoengine_operators::source::GdalMetaDataStatic;
+use geoengine_operators::util::gdal::{
+    gdal_open_dataset_ex, gdal_parameters_from_dataset, raster_descriptor_from_dataset,
+};
+use geoengine_operators::{
+    engine::{
+        MetaData, MetaDataProvider, RasterQueryRectangle, RasterResultDescriptor,
+        VectorQueryRectangle, VectorResultDescriptor,
+    },
+    mock::MockDatasetDataSourceLoadingInfo,
+    source::{GdalLoadingInfo, OgrSourceDataset},
+};
+use log::info;
+use reqwest::Client;
+use serde::{Deserialize, Serialize};
+use snafu::ResultExt;
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Natur40DataProviderDefinition {
```

Comment:
I would recommend renaming the file and the providers to *Nature40*, cf. https://www.uni-marburg.de/en/fb19/natur40/about-us/nature-4-0-in-a-nutshell

Reply:
1944f13ce4342297fba7bfcb43637b0ffe59e3a4